docker/api/Dockerfile

```diff
@@ -46,10 +46,12 @@ RUN chown -R www-data:www-data /tmp/devicecerts && \
 # Give permission for unittests
 RUN chown root:www-data /opt/cnaas/*.sh && \
     chmod g+rx /opt/cnaas/*.sh
-RUN chown -R www-data:www-data /opt/cnaas/venv/cnaas-nms/src/
+RUN chown -R www-data:www-data /opt/cnaas/venv/
```

Comment:
Why is this needed?

Reply:
`cnaas-setup-branch.sh` does two things:

1. checkout the branch
2. install requirements via pip

For 1., it needs permissions on .git.
For 2., it needs permissions on the venv.

If www-data cannot write to the venv, execution of the script fails and `docker-compose build` exits with an error.

Reply:
Ah ok that makes sense. But in this case I think the cnaas-setup-branch.sh should be run as root, and then we change permissions after so it matches what we do for non-branch builds.
I think this is the relevant chown things we do in the main docker thing:
chown -R root:www-data /opt/cnaas/venv/cnaas/
chmod -R u=rwX,g=rX,o= /opt/cnaas/venv/cnaas/
chown -R www-data:www-data /opt/cnaas/venv/cnaas-nms/src/
Ideally the last thing where we change owner of src/ to www-data should not be needed, it says in the comment it's for unit tests but I can't remember why it's needed. Maybe just for creating the .coverage files

Reply:
I made a branch to test this out in bugfix.www-data_permissions_branchesandnosetests and also changed so www-data is not permanently the owner of src/ and it seems to work at my end at least